packages/components/src/FilterBar/context/reducer/filtersStateReducer.ts

```diff
@@ -0,0 +1,24 @@
+import { FilterState, FiltersState, FiltersValues } from "../types"
+import { updateSingleFilter } from "./updateSingleFilter"
+import { updateValues } from "./updateValues"
+
+type Actions<ValuesMap> =
+  | { type: "update_values"; values: Partial<ValuesMap> }
+  | {
+      type: "update_single_filter"
+      id: keyof ValuesMap
+      data: Partial<FilterState<keyof ValuesMap, ValuesMap[keyof ValuesMap]>>
+    }
+
+export const filtersStateReducer = <ValuesMap extends FiltersValues>(
+  state: FiltersState<ValuesMap>,
+  action: Actions<ValuesMap>
+): FiltersState<ValuesMap> => {
+  switch (action.type) {
```

Comment:
🤔 Just wondering, do `switch` statements not mandate a `default` option? Maybe that was just a linting rule in the last code base I worked in. Anyway ...

Reply:
Not if you have TypeScript :D 
I could put a `default` in, but it's something we'd never reach without TS erroring on us so 🤷🏻